EmpireLegends/GameStart.cpp

```diff
@@ -0,0 +1,198 @@
+			std::cout << "INVALID selection." << std::endl;
+			std::cout << "Please enter the type of map you wish to play with." << std::endl;
+			std::cout << "	- Enter 1 for a rectangular map (4 continents)." << std::endl;
+			std::cout << "	- Enter 2 for an L-shaped map (3 continents)." << std::endl;
+			std::cout << "	- Enter 3 for an T-shaped map (4 continents)." << std::endl;
+			std::cin >> mapType;
+		}
+	}
+}
+
+void GameStart::detPlayerCount() {
+
+	int playerCoins = 0;
+
+	std::cout << "How many players will be playing? Please select a number between 2 and 4: ";
+	std::cin >> numOfPlayers;
+
+	// Assign coins according to numofPlayers
+	switch (numOfPlayers) {
+	case 2:
+
+		playerCoins = 14;
+
+		std::cout << "Enter player name 1: ";
+		std::cin.ignore();
+		getline(std::cin, playerName1);
+		p1 = new Player(playerName1, playerCoins);
+		std::cout << "Welcome " << playerName1 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 2: ";
+		getline(std::cin, playerName2);
+		p2 = new Player(playerName2, playerCoins);
+		std::cout << "Welcome " << playerName2 << " !" << std::endl;
+
+		std::cout << "\nEach player gets 14 coins." << std::endl;
+
+		break;
+
+	case 3:
+
+		playerCoins = 11;
+
+		std::cout << "Enter player name 1: ";
+		std::cin.ignore();
+		getline(std::cin, playerName1);
+		p1 = new Player(playerName1, playerCoins);
+		std::cout << "Welcome " << playerName1 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 2: ";
+		getline(std::cin, playerName2);
+		p2 = new Player(playerName2, playerCoins);
+		std::cout << "Welcome " << playerName2 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 3: ";
+		getline(std::cin, playerName3);
+		p3 = new Player(playerName3, playerCoins);
+		std::cout << "Welcome " << playerName3 << " !" << std::endl;
+
+		std::cout << "\nEach player gets 11 coins." << std::endl;
+
+		break;
+
+	case 4:
+
+		playerCoins = 9;
+
+		std::cout << "Enter player name 1: ";
+		std::cin.ignore();
+		getline(std::cin, playerName1);
+		p1 = new Player(playerName1, playerCoins);
+		std::cout << "Welcome " << playerName1 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 2: ";
+		getline(std::cin, playerName2);
+		p2 = new Player(playerName2, playerCoins);
+		std::cout << "Welcome " << playerName2 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 3: ";
+		getline(std::cin, playerName3);
+		p3 = new Player(playerName3, playerCoins);
+		std::cout << "Welcome " << playerName3 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 4: ";
+		getline(std::cin, playerName4);
+		p4 = new Player(playerName4, playerCoins);
+		std::cout << "Welcome " << playerName4 << " !" << std::endl;
+
+		std::cout << "\nEach player gets 9 coins." << std::endl;
+
+		break;
+
+	default:
+		while (numOfPlayers > 4 || numOfPlayers < 2) {
+			std::cout << "INVALID number of players." << std::endl;
+			std::cout << "Please select a number between 2 and 4: ";
+			std::cin >> numOfPlayers;
+		}
```

Comment:
shouldn't this loop be before the switch?

Reply:
i thought it was better placed on the "default" case of the switch case, wherein if the number isn't in the [2, 4] range it'll loop but now that I think about it we may not need the "while" loop at all? since in all cases not caught by the switch (which ends up being the same as the "while" loop conditio, it'll just prompt the user for another input?